### PR TITLE
Restore auto generation of compiler option table

### PR DESCRIPTION
**Pull request by zliu41**
It has been hardcoded since switching from readthedocs to docusaurus.

### CODE REVIEW

File: doc/docusaurus/docs/reference/plutus-tx-compiler-options.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 sidebar_position: 5
```
Comment (kwxm) [inline doc/docusaurus/docs/reference/plutus-tx-compiler-options.md:2]: This appears if you look at the rendered markdown on GhiHub. Is it needed to produce the [fancy documentation](https://plutus.cardano.intersectmbo.org/docs/reference/plutus-tx-compiler-options)?

Sidebar

File: doc/docusaurus/docs/reference/plutus-tx-compiler-options.md [hunk 2]
```diff
@@ -13,33 +18,34 @@ These options can be passed to the compiler via the `OPTIONS_GHC` pragma, for in
...
-| `simplifier-beta`                | Bool          | True    | Run a simplification pass that performs beta transformations |
-| `simplifier-inline`              | Bool          | True    | Run a simplification pass that performs inlining |
-| `simplifier-remove-dead-bindings`| Bool          | True    | Run a simplification pass that removes dead bindings |
-| `simplifier-unwrap-cancel`       | Bool          | True    | Run a simplification pass that cancels unwrap/wrap pairs |
-| `strictify-bindings`             | Bool          | True    | Run a simplification pass that makes bindings stricter |
-| `target-version`                 | Version       | 1.1.0   | The target Plutus Core language version |
-| `typecheck`                      | Bool          | True    | Perform type checking during compilation. |
-| `verbosity`                      | Verbosity     | Quiet   | Set logging verbosity level (0=Quiet, 1=Verbose, 2=Debug) |
-
+|Option|Value Type|Default|Description|
```
Comment (kwxm) [inline doc/docusaurus/docs/reference/plutus-tx-compiler-options.md:21]: It's a pity that the nice alignment's gone in the raw markdown now, but that looks as if it'd be quite hard to fix.

File: plutus-tx-plugin/app/GeneratePluginOptionsDoc.hs [hunk 3]
```diff
@@ -40,29 +41,34 @@ main = do
...
-     - Value Type
-     - Default
-     - Description
+# Plutus Tx compiler options
 
+These options can be passed to the compiler via the `OPTIONS_GHC` pragma, for instance
+
+``` haskell
+{{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:dump-uplc #-}}
+{{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:max-simplifier-iterations=3 #-}}
```
Comment (kwxm) [inline plutus-tx-plugin/app/GeneratePluginOptionsDoc.hs:60]: There's no such option!

```suggestion
{{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:max-simplifier-iterations-uplc=3 #-}}
```

**Review by kwxm — APPROVED** [review]
Look good.

File: doc/docusaurus/docs/reference/plutus-tx-compiler-options.md (hunk 1 above)
Comment (zliu41) [reply doc/docusaurus/docs/reference/plutus-tx-compiler-options.md:2]: Yes

File: doc/docusaurus/docs/reference/plutus-tx-compiler-options.md (hunk 2 above)
Comment (zliu41) [reply doc/docusaurus/docs/reference/plutus-tx-compiler-options.md:21]: Yeah but the raw markdown has always been unreliable.

File: plutus-tx-plugin/app/GeneratePluginOptionsDoc.hs (hunk 3 above)
Comment (zliu41) [reply plutus-tx-plugin/app/GeneratePluginOptionsDoc.hs]: Good catch!

**Pull request merged**